File: Source/Fantasy.Jobs/Resources/JobHostCountResourceProvider.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Fantasy.Jobs.Management;

namespace Fantasy.Jobs.Resources
{
    public class JobHostCountResourceProvider : ConcurrentCountResourceProviderBase
    {
       

        public override bool CanHandle(string name)
        {
            return string.Equals("RunJob", name, StringComparison.OrdinalIgnoreCase);
        }

        public override void Initialize()
        {
            
            JobManagerSettings.Default.PropertyChanged += new System.ComponentModel.PropertyChangedEventHandler(SettingsChanged);
        }

        void SettingsChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
        {
            if (e.PropertyName == "JobProcessCount")
            {
                this.TryRevoke();
            }
            this.OnAvailable();
        }


        protected override int  GetMaxCount(string key)
        {
 	        return JobManagerSettings.Default.JobProcessCount; 
        }

        protected override string GetKey(ResourceParameter parameter)
        {
            return string.Empty;
        }
       

       
    }
}

File: AutoConverted/Fantasy.Jobs/Resources/JobHostCountResourceProvider.java
package Fantasy.Jobs.Resources;

import Fantasy.Jobs.Management.*;

public class JobHostCountResourceProvider extends ConcurrentCountResourceProviderBase
{


	@Override
	public boolean CanHandle(String name)
	{
		return String.equals("RunJob", name, StringComparison.OrdinalIgnoreCase);
	}

	@Override
	public void Initialize()
	{

//C# TO JAVA CONVERTER TODO TASK: Java has no equivalent to C#-style event wireups:
		JobManagerSettings.Default.PropertyChanged += new System.ComponentModel.PropertyChangedEventHandler(SettingsChanged);
	}

	private void SettingsChanged(Object sender, System.ComponentModel.PropertyChangedEventArgs e)
	{
		if (e.PropertyName.equals("JobProcessCount"))
		{
			this.TryRevoke();
		}
		this.OnAvailable();
	}


	@Override
	protected int GetMaxCount(String key)
	{
		 return JobManagerSettings.getDefault().getJobProcessCount();
	}

	@Override
	protected String GetKey(ResourceParameter parameter)
	{
		return "";
	}



}